### ISSUE TITLE
Returning an uninitialised register fails

### ISSUE THREAD

**Issue by haselwarter** [jasmin-lang/jasmin#80]
The program
```
export fn main () -> reg u64 {
    reg u64 res;
    return res;
}
```
fails to compile with the error message
```
"", line 1 (0) to line 4 (1):
internal compilation error in function main:
  one-varmap checker: not able to ensure equality of the result
Please report at https://github.com/jasmin-lang/jasmin/issues
```

The compiler should probably fail more gracefully.

**Comment by eponier** [comment]
Thanks for the report! Actually, this is a known bug, and sadly it is also a regression. See https://github.com/jasmin-lang/jasmin/issues/56#issuecomment-915245725. The errors were not completely clear before, but much clearer than today.

**Comment by vbgl** [comment]
The compiler did not do anything wrong here (so the “please report” is unfortunate).

Here is a more interesting example (it is safe but the compiler exhibits the same behavior):

~~~
export
fn f(reg u64 input) -> reg u64 {
  reg u64 result;
  reg bool b;
  b = input == 0;
  if b { result = 0; }
  if !b { result = 1; }
  return result;
}
~~~

Should we update the error message to some apology: “sorry, proving that all variables are initialized is too challenging in your program.” (and include variable names if possible)?

**Comment by vbgl** [comment]
> I don't know what the ideal error message looks like, but it certainly mentions uninitialized variables.

In my example, compilation MUST succeed.

**Comment by eponier** [comment]
I don't understand, you mention that something is too challenging for the compiler, but then claim it should be compiled?

**Comment by eponier** [comment]
I must admit I don't know what the problem is with these uninitialized variables to begin with.

**Comment by vbgl** [comment]
My bad. I thought you asked for the “ideal” situation. Short term, indeed, let’s improve the error message.

**Comment by vbgl** [comment]
Fixed by 0e9e2fa42ff63bbf8d33720e3ec6c7436163f72d.
